### ISSUE TITLE
Ollama hangs with multi-modal models

### ISSUE THREAD

**Issue by jmorganca** [ollama/ollama#3361]
### What is the issue?

```
Apr 04 05:15:04 gpu.us-central1-a.c.ollama.internal ollama[5042]: {"function":"launch_slot_with_data","level":"INFO","line":804,"msg":"slot is processing task","slot_id":0,"task_id":29930,"tid":"140079034640064","timestamp":1712207704}
Apr 04 05:15:04 gpu.us-central1-a.c.ollama.internal ollama[5042]: {"function":"update_slots","level":"INFO","line":1808,"msg":"kv cache rm [p0, end)","p0":0,"slot_id":0,"task_id":29930,"tid":"140079034640064","timestamp":1712207704}
Apr 04 05:15:44 gpu.us-central1-a.c.ollama.internal ollama[5042]: [1712207744] update_slots : failed to find free space in the KV cache, retrying with smaller n_batch = 256
Apr 04 05:15:44 gpu.us-central1-a.c.ollama.internal ollama[5042]: [1712207744] update_slots : failed to find free space in the KV cache, retrying with smaller n_batch = 128
Apr 04 05:15:44 gpu.us-central1-a.c.ollama.internal ollama[5042]: [1712207744] update_slots : failed to find free space in the KV cache, retrying with smaller n_batch = 64
Apr 04 05:15:44 gpu.us-central1-a.c.ollama.internal ollama[5042]: [1712207744] update_slots : failed to find free space in the KV cache, retrying with smaller n_batch = 32
Apr 04 05:15:44 gpu.us-central1-a.c.ollama.internal ollama[5042]: [1712207744] update_slots : failed to find free space in the KV cache, retrying with smaller n_batch = 16
Apr 04 05:15:44 gpu.us-central1-a.c.ollama.internal ollama[5042]: [1712207744] update_slots : failed to find free space in the KV cache, retrying with smaller n_batch = 8
Apr 04 05:15:44 gpu.us-central1-a.c.ollama.internal ollama[5042]: [1712207744] update_slots : failed to find free space in the KV cache, retrying with smaller n_batch = 4
Apr 04 05:15:44 gpu.us-central1-a.c.ollama.internal ollama[5042]: [1712207744] update_slots : failed to find free space in the KV cache, retrying with smaller n_batch = 2
Apr 04 05:15:44 gpu.us-central1-a.c.ollama.internal ollama[5042]: [1712207744] update_slots : failed to find free space in the KV cache, retrying with smaller n_batch = 1
Apr 04 05:15:44 gpu.us-central1-a.c.ollama.internal ollama[5042]: [1712207744] update_slots : failed to decode the batch, n_batch = 1, ret = 1
Apr 04 05:15:44 gpu.us-central1-a.c.ollama.internal ollama[5042]: [1712207744] update_slots : failed to find free space in the KV cache, retrying with smaller n_batch = 256
```

**Comment by rainux** [comment]
Same here. I'd like to help debug this issue. Do you have any suggestions? Thanks.

**Comment by jmorganca** [comment]
This should be fixed as of 0.1.31
